src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -34,223 +158,176 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
     {
-        handler;
-
-        throw hresult_not_implemented();
+        return _disconnectHandlers.add(handler);
     }
 
     void ConptyConnection::TerminalDisconnected(winrt::event_token const& token) noexcept
     {
-        token;
-
-        // throw hresult_not_implemented();
+        _disconnectHandlers.remove(token);
     }
 
     void ConptyConnection::Start()
+    try
     {
-        _CreatePseudoConsole();
+        COORD dimensions{ gsl::narrow_cast<SHORT>(_initialCols), gsl::narrow_cast<SHORT>(_initialRows) };
+        THROW_IF_FAILED(_CreatePseudoConsoleAndPipes(dimensions, 0, &_inPipe, &_outPipe, &_hPC));
+        THROW_IF_FAILED(_LaunchAttachedClient());
 
-        _connected = true;
+        _startTime = std::chrono::high_resolution_clock::now();
 
         // Create our own output handling thread
-        // Each console needs to make sure to drain the output from it's backing host.
-        _outputThreadId = (DWORD)-1;
-        _hOutputThread = CreateThread(nullptr,
-                                      0,
-                                      StaticOutputThreadProc,
-                                      this,
-                                      0,
-                                      &_outputThreadId);
-
-        //// When we recieve some data:
-        //hstring outputFromConpty = L"hello world";
-        ////TerminalOutputEventArgs args(outputFromConpty);
-        ////_outputHandlers(*this, args);
-        //_outputHandlers(outputFromConpty);
-    }
-
-    void ConptyConnection::WriteInput(hstring const& data)
-    {
-        data;
+        // This must be done after the pipes are populated.
+        // Each connection needs to make sure to drain the output from its backing host.
+        _hOutputThread.reset(CreateThread(
+            nullptr,
+            0,
+            [](LPVOID lpParameter) {
+                ConptyConnection* const pInstance = reinterpret_cast<ConptyConnection*>(lpParameter);
+                return pInstance->_OutputThread();
+            },
+            this,
+            0,
+            nullptr));
+
+        THROW_LAST_ERROR_IF_NULL(_hOutputThread);
+
+        _clientExitWait.reset(CreateThreadpoolWait(
+            [](PTP_CALLBACK_INSTANCE /*callbackInstance*/, PVOID context, PTP_WAIT /*wait*/, TP_WAIT_RESULT /*waitResult*/) {
+                ConptyConnection* const pInstance = reinterpret_cast<ConptyConnection*>(context);
+                pInstance->_ClientTerminated();
+            },
+            this,
+            nullptr));
+
+        SetThreadpoolWait(_clientExitWait.get(), _piClient.hProcess, nullptr);
 
-        throw hresult_not_implemented();
+        _connected = true;
     }
-
-    void ConptyConnection::Resize(uint32_t rows, uint32_t columns)
+    catch (...)
     {
-        rows;
-        columns;
+        const auto hr = wil::ResultFromCaughtException();
+        // TODO GH#2563 - signal a transition into failed state here!
+        (void)hr;
```

Comment:
Log? or something? Or just put it straight to void instead of UNREFERENCED_PARAMETER-ing it?